research/modeling/bert.py

```diff
@@ -0,0 +1,291 @@
+import tensorflow as tf
+import pandas as pd
+import tensorflow_hub as hub
+import os
+import re
+import numpy as np
+from bert.tokenization import FullTokenizer
+from tqdm import tqdm_notebook
+from tensorflow.keras import backend as K
+
+# Initialize session
+sess = tf.Session()
+
+# Params for bert model and tokenization
+bert_path = "https://tfhub.dev/google/bert_uncased_L-12_H-768_A-12/1"
+max_seq_length = 256
+
+# Load all files from a directory in a DataFrame.
+def load_directory_data(directory):
+  data = {}
+  data["sentence"] = []
+  data["sentiment"] = []
+  for file_path in os.listdir(directory):
+    with tf.gfile.GFile(os.path.join(directory, file_path), "r") as f:
+      data["sentence"].append(f.read())
+      data["sentiment"].append(re.match("\d+_(\d+)\.txt", file_path).group(1))
+  return pd.DataFrame.from_dict(data)
+
+# Merge positive and negative examples, add a polarity column and shuffle.
+def load_dataset(directory):
+  pos_df = load_directory_data(os.path.join(directory, "pos"))
+  neg_df = load_directory_data(os.path.join(directory, "neg"))
+  pos_df["polarity"] = 1
+  neg_df["polarity"] = 0
+  return pd.concat([pos_df, neg_df]).sample(frac=1).reset_index(drop=True)
```

Comment:
add comment explaining that sample(frac=1) is a random permutation. should add injection of random_state for determinism during testing.